doc/src/kubernetes.md

```diff
@@ -173,120 +173,182 @@ VMs and all services managed by our platform can be accessed as usual by
-to terminate SSL and virtual host configuration within Kubernetes.
+to terminate SSL and virtual host configuration within Kubernetes using an
+ingress controller like traefik, which is installed by default.
+
+The setup in Kubernetes looks like this:
+
+- Deploy an ingress controller or use traefik ingress installed by default
+  (used in the example below).
+- Bind the ingress controller to a load balancer (port 8888 in our example).
+- Add a service of type ClusterIP for your application.
+- Deploy an ingress resource to to pass traffic to your application.
+
+On the platform side, we have:
+
+- HAProxy, listening on a public interface, forwarding requests to an ingress
+  controller service.
+
 
 Here's a diagram of the setup focusing on your application's traffic flow:
 
 ```{image} images/tcp_ingress.png
 ```
 
-The setup in Kubernetes looks like this:
-
-- expose your application's port using a service
-- (bind your application's port to a load balancer)
-- deploy an ingress controller and configure it to pass traffic to your
-  application (port 8443 in our example)
-- bind the ingress controller to a load balancer (port 8888 in our example)
-
-In our platform you connect the ingress load balancer with our webgateway:
+To connect the ingress controller service with our webgateway, both for HTTP
+and HTTPS:
 
 ```nix
 # /etc/local/nixos/myappfrontend.nix
 { ... }:
 {
-
-  flyingcircus.kubernetes.frontend.myapp = {
-    # haproxy now listens on a public IP address
-    bind = [ "192.0.124.96:443" ];
+  flyingcircus.kubernetes.frontend.traefik-ingress-https = {
+    # HAProxy listens on a public IP address using the standard HTTPS port 443
+    binds = [ "192.0.124.96:443" ];
     mode = "tcp"
 
-    # port the ingress load balancer service is listening on
-    lbServicePort = 8888;
-    # port the ingress controller is exposed on the clusterIP of the pod.
-    podPort = 8443;
+    # port the ingress controller's load balancer service is listening on
+    # for HTTPS traffic.
+    lbServicePort = 443;
+    serviceName = "traefik";
+    namespace = "kube-system";
   }
 
+  flyingcircus.kubernetes.frontend.traefik-ingress-http = {
+    binds = [ "192.0.124.96:80" ];
+    mode = "tcp"
+
+    # port the ingress controller's load balancer service is listening on
+    # for HTTP traffic
+    lbServicePort = 80;
+    serviceName = "traefik";
+    namespace = "kube-system";
+  }
 }
 ```
 
+Note that we are using *tcp* mode of HAProxy here which just passes the TCP
+stream to the ingress pods. Load balancing is done inside of Kubernetes. The
+ingress controller is responsible for terminating SSL, managing certificates
+and redirecting from HTTP to HTTPS if applicable. This is different from
+scenario 1 where HAProxy can do the load balancing in front of the Kubernetes
+cluster and nginx on the frontend VM has the responsibilities that the
+ingress controller has in this scenario.
+
+
```

Comment:
There is no szeraio 3 covering the use of `servicePort`, are these szenarios simple enough that there is no need for such an illustrative szenario description?

Reply:
I think that "scenario 3" is not very relevant and I wouldn't recommend it, but that may change with experience ;) It would be very similar to the first scenario and my main intention was to show the two different ways how haproxy can be used in the Kubernetes context.